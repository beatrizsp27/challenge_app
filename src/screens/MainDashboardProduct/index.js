// eslint-disable-next-line
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { k } from '../../utils';
import { GET_PRODUCT_SEARCH} from "../../api";
import { getMessageError } from '../../api/config';
import { ContainerComponent } from "../../component";
import SearchComponent from "../../component/SearchComponent";
import {icShipping} from "../../assets";

const MainDashboardProduct = () => {

	const navigate = useNavigate();
	const [errorMessage, setErrorMessage] = useState('');
	const [arrayListProduct, setArrayListProduct] = useState([]);

	const getProductSearch = async (text) =>{
		console.log("text.target.value_-----------"+ text);
		if(text  && text.target && text.target.value){
			console.log("text.target.value"+ text.target);
			console.log("text.target.value"+ text.target.value);
			await GET_PRODUCT_SEARCH(
				'iphone',
				onSuccessGetProductSearch,
				onErrorGetProductSearch,
				onDoneGetProductSearch
			);
		}else {
			setArrayListProduct([]);
		}
	};

	const onSuccessGetProductSearch = response =>{
		const { data, success } = response.data;
		if (success) {
			if(data){
				const {items} = data;
				console.log("items" + JSON.stringify(items))
				setArrayListProduct(items);
			}
		}
	};

	const onErrorGetProductSearch = error =>{
		if (error && error.response && error.response.data) {
			const {
				response: {
					data: { message }
				}
			} = error;
			showError(message || getMessageError(error));
		} else {
			showError(getMessageError(error));
		}
	};

	const onDoneGetProductSearch = () =>{
		setErrorMessage(null);
	}

	const showError = message => {
		setErrorMessage(message);
	};

	const goDetailsProduct = (id) => {
		navigate(`${k.navigations.detailsProduct}${id}`);
	}

	return (
		<div>
			<SearchComponent handleChange={(text)=>getProductSearch(text)}/>
			<>
				{
					errorMessage &&  (
						<h1>{errorMessage}</h1>
					)
				}
				{arrayListProduct && arrayListProduct.length > 0 && (
					<>
						<div className={'screens_title_body_header'}>
							<label  className={'screens_title_header'}>{'Electronica > audio > video > ipno'} </label>
						</div>
						<ContainerComponent>
							{arrayListProduct.map((item)=> {
								if(item){
									return (
										<div onClick={() => goDetailsProduct(item.id)} key={item.id} className={'screens_card'}>
											<div className={'screens_img'}>
												<img  src={item.picture} alt={item.title} width="180px" height="180px"/>
											</div>
											<div className={'screens_container_card'}>
												{item.price && item.price.decimals && (
													<div className={'screens_card_icon_title'}>
														<h3 className={'text_price'} >{`$ ${item.price.decimals}`} </h3>
														{item.free_shipping && (
															<img src={icShipping} alt={item.title} width="25px"
																 height="25px"/>
														)}
													</div>
												)}
												<>
													{item.title && (
														<label className={'screens_title'}>{item.title}</label>
													)}
												</>
											</div>
											<div className={'screens_card_row3'}>
												<h5 className={'screens_subtitle'} >{item.free_shipping ? 'Capital Federal' : 'Mendoza'}</h5>
											</div>
										</div>
									)
								}
								return null;
							})
							}
						</ContainerComponent>
					</>
				)}
			</>
		</div>

	);
};

export default MainDashboardProduct;
