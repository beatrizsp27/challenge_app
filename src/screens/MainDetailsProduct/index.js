import React, {useState,useEffect} from 'react';
import { useNavigate , useParams} from 'react-router-dom';
import { getMessageError } from '../../api/config';
import {GET_PRODUCT_BY_ID} from '../../api'


const MainDetailsProduct = () => {
	/**HOOKS GENERALES*/
	const navigate = useNavigate();
	const params = useParams();

	/**HOOKS VARIABLES*/
	const [errorMessage, setErrorMessage] = useState('');
	const [loader, setIsLoader] = useState(false);
	const [product, setProduct] = useState(null);

	const goBack = () => {
		navigate(-1);
	};

	useEffect(()=>{
		if (params && params.code) {
			setIsLoader(true);
			const { id } = params;
			getDetailsProduct(id)
		}
	}, [])


	const getDetailsProduct  = (id) => {
		GET_PRODUCT_BY_ID(
			id,
			onSuccessGetDetailsProduct,
			onErrorGetDetailsProduct,
			onDoneGetDetailsProduct
		)
	};

	const onSuccessGetDetailsProduct = response =>{
		console.log(response)
		const { data, success } = response.data;
		if (success) {
			if(data){
				const {item} = data;
				setProduct(item);
			}
		}
	};

	const onErrorGetDetailsProduct = error =>{
		console.log(error)
		if (error && error.response && error.response.data) {
			const {
				response: {
					data: { message }
				}
			} = error;
			showError(message || getMessageError(error));
		} else {
			showError(getMessageError(error));
		}
	};

	const onDoneGetDetailsProduct = () =>{
		setErrorMessage(null);
	}

	const showError = message => {
		setErrorMessage(message);
	};

	return (
		<>
			{
				errorMessage &&  (
					<h1>{errorMessage}</h1>
				)
			}
			<div>
				<h1>{product.title} </h1>
				<button onClick={() => goBack()}>Regresar</button>
			</div>
		</>
	);
};

export default MainDetailsProduct;
