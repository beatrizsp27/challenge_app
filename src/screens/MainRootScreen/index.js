// eslint-disable-next-line
import React from 'react';

const MainSearchProduct = () => {
    return (
        <div/>
    );
};

export default MainSearchProduct;
