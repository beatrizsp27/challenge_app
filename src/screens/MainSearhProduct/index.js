// eslint-disable-next-line
import React, { useState } from 'react';
import {SearchComponent} from "../../component";
import {k} from "../../utils";
import {useNavigate} from "react-router-dom";

const MainSearchProduct = () => {

    const [textSearch, setTextSearch] = useState(null);
    const navigate = useNavigate();
    const handleChange = (text) =>{
        if(text && text.target && text.target.value){
            setTextSearch(text.target.value);
        }else {
            setTextSearch(null);
            navigate('/');
        }
    };

    const goToNavigateProduct = ()=>{
        navigate(`${k.navigations.productItem}?search=${textSearch}`)
        window.location.reload();
    };

    return (
        <div>
            <SearchComponent
                textSearch={textSearch}
                onAction={goToNavigateProduct}
                handleChange={(text)=>handleChange(text)}/>
        </div>
    );
};

export default MainSearchProduct;
