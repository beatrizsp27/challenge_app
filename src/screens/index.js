export { default  as MainDashboardProductScreen } from './MainDashboardProduct';
export { default  as MainDetailsProductScreen } from './MainDetailsProduct';