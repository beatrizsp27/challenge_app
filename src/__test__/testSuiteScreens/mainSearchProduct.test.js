import React from "react";
import {MainDashboardProductScreen, MainSearchProductScreen} from '../../screens';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import {shallow, configure} from "enzyme";
configure({ adapter: new Adapter() })

describe("MainSearchProductScreen", ()=>{

    /** SE DECLARA UNA VARABLES**/
    let wrapperMainSearchProductScreen;

    beforeAll(()=>{
        wrapperMainSearchProductScreen = shallow
        (<Router>
            <MainSearchProductScreen/>
        </Router>);
    });

    /** SE RENDERIZA EL COMPONENTE CASO DE PRUEBA**/
    test("renderear correctamente", () => {
        /** ESPERANDO QUE WAPER ESTE DEFENIDO Y QUE EXISTA**/
        expect(wrapperMainSearchProductScreen).toBeTruthy()

    })

    /** SE RENDERIZA EL COMPONENTE CASO DE PRUEBA**/
    test("Se crea el snap", () => {
        const tree = renderer.create(
            <Router>
                <MainSearchProductScreen/>
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});


