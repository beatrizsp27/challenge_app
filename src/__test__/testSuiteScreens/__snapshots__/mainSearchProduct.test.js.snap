// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearchProductScreen Se crea el snap 1`] = `
<div
  className="screens_details_header"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="screens_details_form"
    >
      <div
        className="box_img_search"
      >
        <img
          className="box_img"
          src="Logo_ML@2x.png.png"
        />
      </div>
      <div
        className="box_input_icon_search"
      >
        <input
          className="box_input_text"
          name="search"
          onChange={[Function]}
          placeholder="Nunca dejes de buscar"
          type="search"
        />
        <div
          className="box_search_icon"
          onClick={[Function]}
        >
          <img
            alt=""
            className="box_img_mel"
            src="ic_Search@2x.png.png"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
